### ISSUE TITLE
Settings updating for production sites

### ISSUE THREAD

**Issue by ferblape** [consul/consul#1362]
Morning!

When you introduce big features such as the budgets, there are always a few settings involved that are necessary to enable the feature and configure it properly. They are easy to find in the `db/seeds.rb` file.

But, if a site is already in production, and you run the seeds, the settings will be overwritten, which is something we should try to avoid.

What if we change the `db/seeds.rb` file to use a syntax like this?

```ruby
...
# Feature flags
Setting['feature.debates'] ||= true
Setting['feature.spending_proposals'] ||= nil
Setting['feature.twitter_login'] ||= true
Setting['feature.facebook_login'] ||= true
Setting['feature.google_login'] ||= true
Setting['feature.public_stats'] ||= true
Setting['feature.budgets'] ||= true
Setting['feature.signature_sheets'] ||= true
...
```

Would you accept this in a PR?

**Comment by xuanxu** [comment]
We like the idea of making seeds.rb idempotent, but using `||=` won't be enough, because there are some settings with nil value on purpose. I think it would be better to use a specific method, someting like:

````ruby
Setting.exists_or_default_value('feature.debates', true)
````
And then in the Setting model 
````ruby
def self.exists_or_default_value(key, default)
  # creates the setting with the default value only if it doesn't exists, 
  #so existent settings with nil value are maintained nil
end
````
(`exists_or_default_value` is an ugly name, but you get the idea :)
WDYT?

**Comment by javierm** [comment]
> We like the idea of making seeds.rb idempotent

@xuanxu Agreed :+1:. A first step is being done in AyuntamientoMadrid#1746. Once we've refactored this part, making `seeds.rb` idempotent should be easy, and there's a draft at javierm#6.

**Comment by javierm** [comment]
Closed by #3576.
